src/api/app/models/project.rb

```diff
@@ -1391,25 +1392,22 @@ def release_targets_ng
           package_issues: {},
           package_issues_by_tracker: {}
         }
+        repo_name_to_project_name[repo.name] = rt.target_repository.project.name
       end
     end
 
     # One catch, currently there's only one patchinfo per incident, but things keep changing every
     # other day, so it never hurts to have a look into the future:
     package_count = 0
-    packages.where.not(id: global_patchinfo_package).select(:name, :id).each do |pkg|
+    packages.where.not(id: global_patchinfo_package).where("name like '%.%'").select(:name, :id).each do |package|
       # Current ui is only showing the first found package and a symbol for any additional package.
       break if package_count > 2
-
-      rt_name = pkg.name.split('.', 2).last
-      next unless rt_name
-      # Here we try hard to find the release target our current package is build for:
-      rt_name = guess_release_target_from_package(pkg, release_targets_ng)
-
-      # Build-disabled packages can't be matched to release targets....
-      next unless rt_name
+      # Here we try to find the release target our current package is build for:
+      first_matching_build_target = package.flags.where(flag: :build, status: 'enable', repo: repo_name_to_project_name.keys).select(:repo).first
```

Comment:
careful. not disabled doesn't mean enabled - enable is the default and doesn't have a flag entry then. But in case of incidents there seems to be explicit flags.

But https://build.opensuse.org/package/meta/openSUSE:Maintenance:9588/man-pages-posix.openSUSE_Leap_15.0_Update makes me wonder what significance the first repo has. The first seems to be _ports, but the release target 'from package' would be _Update.

Reply:
You can read in every single comment just how frustrated sascha was about all this :)

Reply:
> careful. not disabled doesn't mean enabled - enable is the default and doesn't have a flag entry then. But in case of incidents there seems to be explicit flags.

I found that strange as well. Though in the maintenance incident packages on OBS you can see that they explicitly enable the package flags.